lib/std/crypto/25519/scalar.zig

```diff
@@ -150,11 +163,51 @@ pub inline fn clamp(s: *[32]u8) void {
     s[31] = (s[31] & 127) | 64;
 }
 
+/// Return a*b (mod L)
+pub fn mul(a: [32]u8, b: [32]u8) [32]u8 {
+    return ScalarExpanded.fromBytes(a).mul(ScalarExpanded.fromBytes(b)).toBytes();
+}
+
 /// Return a*b+c (mod L)
 pub fn mulAdd(a: [32]u8, b: [32]u8, c: [32]u8) [32]u8 {
     return ScalarExpanded.fromBytes(a).mulAdd(ScalarExpanded.fromBytes(b), ScalarExpanded.fromBytes(c)).toBytes();
 }
 
+/// Return a*8 (mod L)
+pub fn mul8(s: [32]u8) [32]u8 {
```

Comment:
This seems a little... specific.
With only a single call site, why have this at all?
Alternatively, what if this was generic over some comptime known multiple?

Reply:
This is very specific. 8 is the cofactor of the groups these scalars represent elements of.

We only have one use case here, but multiplying a secret scalar per 8 is a common operation when using these curves. 

A similar operation is exposed as `clearCofactor()` in `edwards25519` and `curve25519`, but I didn't want to use that terminology for a finite field, where talking about cofactors wouldn't make sense.